pkg/console/subresource/managedserviceproxyresolver/apply.go

```diff
@@ -0,0 +1,48 @@
+package managedproxyserviceresolver
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+	clusterproxy "open-cluster-management.io/cluster-proxy/pkg/apis/proxy/v1alpha1"
+	proxyclient "open-cluster-management.io/cluster-proxy/pkg/generated/clientset/versioned/typed/proxy/v1alpha1"
+)
+
+func ApplyManagedProxyServiceResolver(ctx context.Context, client proxyclient.ManagedProxyServiceResolverInterface, required *clusterproxy.ManagedProxyServiceResolver) error {
+	existing, err := client.Get(ctx, required.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		_, err = client.Create(ctx, required, metav1.CreateOptions{})
+		if err != nil {
+			klog.V(4).Infof("faied to create thanos-querier ManagedServiceProxyResolver: %s/n", err)
+			return err
+		}
+		return nil
```

Comment:
return will trigger another loop. we could just continue if no error occurs

Reply:
I'm not sure I follow. This func only returns an error. If it returns nil, that means the required resource was successfully applied and no further action is required from the controller.

Reply:
hmm looking at my comment now it seems I've already had christmas spirit, while writing 😄  please ignore